hibike/lib/hibike/util.cpp

```diff
@@ -0,0 +1,20 @@
+#include "util.h"
+
+// Assumes little endian
+
+uint64_t read_num_bytes(uint8_t* data, size_t num) {
+  uint64_t value = 0;
+  for (size_t i = 0; i < num; i++) {
+    value += data[i] << (8 * i);
+  }
+  return value;
+}
+
+
+// Assumes little endian
+
+void write_num_bytes(uint64_t data, uint8_t* data_buf, size_t num) {
```

Comment:
If we use this function to write different data types to a buffer, could the cast to uint64_t cause incorrect behavior when dealing with signed integers / floating point numbers? Why not just pass a pointer to data instead of casting and doing bit operations?